spark/spark.scala

```diff
@@ -1,28 +1,40 @@
 import org.apache.log4j._
-Logger.getRootLogger.setLevel(Level.OFF)
-import org.apache.spark.mllib.ann.{FeedForwardTrainer, FeedForwardTopology}
+import org.apache.spark.ml.classification.MultilayerPerceptronClassifier
 import org.apache.spark.mllib.linalg.Vectors
 import org.apache.spark.mllib.regression.LabeledPoint
 import org.apache.spark.mllib.util.MLUtils
-import org.apache.spark.mllib.classification.ANNClassifier
+
+Logger.getRootLogger.setLevel(Level.OFF)
+
 // maximum number of worker nodes in cluster
-val numNodes = 5
-// batch size, ~10K is good for GPU
-val batchSize = 10000
+val numNodes= 5
+
 // number of iterations to run
 val numIterations = 5
-val train = MLUtils.loadLibSVMFile(sc, "/mnist.scale")
-val topology = FeedForwardTopology.multiLayerPerceptron(Array[Int](780, 2500, 2000, 1500, 1000, 500, 10), false)
-val trainer = new FeedForwardTrainer(topology, 780, 10).setBatchSize(batchSize)
-trainer.SGDOptimizer.setNumIterations(numIterations).setMiniBatchFraction(1.0).setStepSize(0.03)
+
+-// block size, ~10K is good for GPU
+-val blockSize = 10000
+
+val trainData = MLUtils.loadLibSVMFile(sc, "data/mllib/mnist.scale")
+val layers = Array[Int](780, 2500, 2000, 1500, 1000, 500, 10)
+val estimator = new MultilayerPerceptronClassifier()
+  .setLayers(layers)
+  .setBlockSize(blockSize)
+  .setMaxIter(numIterations)
+  .setSeed(11L)
+
```

Comment:
The public API of MLP does not allow setting the SGD optimizer. The default optimizer is LBFGS instead. It does not allow direct measuring the throughput as compared to SGD. We should come up with a workaround that allows setting SGD. We might either put the code in the `ann` package, or create a helper class in the private package that will change the optimizer.